src/cosp_stats.F90

```diff
@@ -331,7 +331,10 @@ SUBROUTINE COSP_DIAG_WARMRAIN( Npoints, Ncolumns, Nlevels,           & !! in
     real(wp) :: cbtmh     !! diagnosed in-cloud optical depth
     real(wp), dimension(Npoints,Ncolumns,Nlevels) :: icod  !! in-cloud optical depth (ICOD)
     logical  :: octop, ocbtm, oslwc
-
+    integer, dimension(Npoints,Ncolumns,Nlevels) :: fracout_int  !! fracout (decimal to integer)
+    where (fracout .le. 0.5)                        fracout_int = SGCLD_CLR ! 0 = clear-sky
```

Comment:
Is there some reason not to use the `NINT()` intrinsic here? Also, perhaps you can explain in comments what's happening here.

Reply:
Thanks for this suggestion. I modified here using NINT() intrinsic function. I confirmed that the output result was not changed.